server.js

```diff
@@ -1,28 +1,86 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
+app.get('/', (_, res) => {
   res.send('Hello world')
 })
 
+//Returns 20 sorted thoughts
+app.get('/thoughts', async (_, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
+
+//Adds and saves a thought to the database
+app.post('/thoughts', async (req, res) => {
+  const { message } = req.body
+  const thought = new Thought({message})
+  try {
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {
+    res.status(400).json({message: 'Could not save the thought to the database.', error: err.errors})
+  }
+})
+
+//Adds a like (a heart) to a specifik thought
+app.post('/thoughts/:id/likes', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(
+      id, 
```

Comment:
We were unsure about the formatting here. Guess id can be shortened like that if it's id:id? (but then it's not the mongo-id that's being used?) But it seems to work 😎

Reply:
@katliebig - Linnea used a special kind of function `findByIdAndUpdate` which is by default looking for `_id` field.